fmn/rules/utils.py

```diff
@@ -341,18 +370,8 @@ def _get_pagure_packages_for(config, username, flags):
             # Belt and suspenders.  We should never get here.
             raise NotImplementedError("%r is not a valid flag" % flag)
 
-        try:
-            response = requests.get(url, params=params, timeout=15)
-        except requests.exceptions.ConnectTimeout as e:
-            log.warn('URL %s timed out %r', response.url, e)
-            continue
-
-        if not response.status_code == 200:
-            log.warn('URL %s returned code %s', response.url, response.status_code)
-            continue
-
-        data = response.json()
-        for repo in data['projects']:
+        repos = _paginate_pagure_data(url, params, key='projects')
```

Comment:
Can you add `short=true` to the parameters since you only need the namespace and name of project? It will speed up the queries significantly.